client/blocks/signup-form/index.jsx

```diff
@@ -456,20 +456,38 @@ class SignupForm extends Component {
 		);
 	}
 
-	getUserData() {
-		const extraFields = {
-			extra: {
-				first_name: formState.getFieldValue( this.state.form, 'firstName' ),
-				last_name: formState.getFieldValue( this.state.form, 'lastName' ),
-			},
-		};
+	getUserNameHint() {
+		const email = formState.getFieldValue( this.state.form, 'email' );
+		const firstName = formState.getFieldValue( this.state.form, 'firstName' );
+		const lastName = formState.getFieldValue( this.state.form, 'lastName' );
 
-		return {
-			username: formState.getFieldValue( this.state.form, 'username' ),
+		const emailIdentifier = email.match( /^(.*?)@/ );
+		return ( emailIdentifier && emailIdentifier[ 1 ] ) || `${ firstName }.${ lastName }`;
```

Comment:
Since firstname, lastname are optional, doesn't it make sense to add an `if` to check `this.props.displayNameInput === false`?

Reply:
You know, the email is required so let's stick to it.

Reply:
Sorry for the confusion. What I meant is that we don't have to get the field values for first & last names if `this.props.displayNameInput === false`, hence my proposal for adding the check.